expand.py

```diff
@@ -576,6 +581,11 @@ def expand(cfg, lines, f, isfirst=False):
                         cfg.ddict = rename_index(olddict.copy())
                         cfg.ddict.update(inst)
                         expand(cfg, t[0], f, isfirst)
+                        # Now, within the $$LOOP, we might have done some $$ASSIGNs.
+                        # We need to pull these back into olddict!
+                        for dname in cfg.assigns:
```

Comment:
Is there any guarantee that cfg.assigns only holds assignments from this loop? I think this could also hold assigns from prior to the first loop during the first evaluation of a loop block, though I guess this probably doesn't cause any issues

Reply:
No, I think you have a point here.  assigns needs to be changed to keep track of the nesting level.  (If  we have two loops, the inner would export to the outer, but the outer would just drop the value on the floor when it went back to the top-level namespace.

Reply:
Right that makes sense, I forgot about nested loops